frappe/public/js/frappe/form/controls/link.js

```diff
@@ -517,50 +512,138 @@ frappe.ui.form.ControlLink = class ControlLink extends frappe.ui.form.ControlDat
+						return `"${String(__(v))}"`;
+					}
+					// Numbers, dates, etc. - not translated, not quoted
+					return String(v);
+				});
+				if (filtered.length > 5) arr.push("...");
+				return arr.join(", ");
 			}
 
-			let value;
-			if (filter[3] && Array.isArray(filter[3])) {
-				value = filter[3].map((v) => String(__(v)).bold()).join(", ");
-			} else if (filter[3] == null || filter[3] === "") {
-				value = __("empty").bold();
-			} else {
-				value = String(__(filter[3])).bold();
+			// Null / empty
+			if (val == null || val === "") {
+				return __("empty");
 			}
 
-			return [__(label).bold(), __(filter[2]), value].join(" ");
+			// Format based on type: strings in quotes, numbers/dates not quoted
+			if (typeof val === "string") {
+				return `"${String(__(val))}"`;
+			}
+
+			// Numbers, dates, etc. - not translated, not quoted
+			return String(val);
 		}
 
-		let filter_string = filter_array.map(get_filter_description).join(", ");
+		async function describe_filter(filter) {
+			// expect [doctype, fieldname, operator, value]
+			const _doctype = filter[0];
+			const fieldname = filter[1];
+			const operator = filter[2];
+			let value = filter[3];
+
+			// Ensure metadata is loaded for this doctype before accessing docfield
+			await frappe.model.with_doctype(_doctype, () => {});
+
+			const docfield = frappe.meta.get_docfield(_doctype, fieldname);
+			const label = docfield ? docfield.label : frappe.model.unscrub(fieldname);
+			const fieldtype = docfield ? docfield.fieldtype : null;
+
+			const labelDisplay = `<i>${String(__(label))}</i>`;
+			const valueDisplay = formatValueForDisplay(docfield, value);
+
+			// Handle all operators with translation and interpolation in one call
+			switch (operator) {
+				case "=":
+					if (fieldtype === "Check") {
+						return value == 1
+							? __("{0} is enabled", [labelDisplay])
+							: __("{0} is disabled", [labelDisplay]);
+					}
+					return __("{0} equals {1}", [labelDisplay, valueDisplay]);
+				case "!=":
+					if (fieldtype === "Check") {
+						return value == 1
+							? __("{0} is disabled", [labelDisplay])
+							: __("{0} is enabled", [labelDisplay]);
+					}
+					return __("{0} is not equal to {1}", [labelDisplay, valueDisplay]);
+				case "in":
+					return __("{0} is one of {1}", [labelDisplay, valueDisplay]);
+				case "not in":
+					return __("{0} is not one of {1}", [labelDisplay, valueDisplay]);
+				case "like":
+					return __("{0} contains {1}", [labelDisplay, valueDisplay]);
+				case "not like":
+					return __("{0} does not contain {1}", [labelDisplay, valueDisplay]);
+				case ">":
+					return __("{0} is greater than {1}", [labelDisplay, valueDisplay]);
+				case "<":
+					return __("{0} is less than {1}", [labelDisplay, valueDisplay]);
+				case ">=":
+					return __("{0} is greater than or equal to {1}", [labelDisplay, valueDisplay]);
+				case "<=":
+					return __("{0} is less than or equal to {1}", [labelDisplay, valueDisplay]);
```

Comment:
In list filters, we have special handling for date/time fields. "less than" in time means "before" and "greater than" means "after". The same applies here.

Reply:
@manzoorsofi we have a "Time" field type as well ;)